src/aws-cpp-sdk-core/include/aws/core/utils/stream/AwsChunkedStream.h

```diff
@@ -56,6 +60,14 @@ class AwsChunkedStream {
     return static_cast<size_t>(m_chunkingStream->gcount());
   }
 
+protected:
+  bool ReadImpl(Crt::ByteBuf &buffer) noexcept override {
+    size_t amountToRead = buffer.capacity - buffer.len;
+    const auto bytesRead = BufferedRead(reinterpret_cast<char *>(buffer.buffer), amountToRead);
+    buffer.len += bytesRead;;
+    return bytesRead > 0;
```

Comment:
https://github.com/awslabs/aws-crt-cpp/blob/743ab3891c4fb015a74713b7b631c720ce0ffe8e/include/aws/crt/io/Stream.h#L108-L113
```
                 * @return true if nothing went wrong.
                 * Return true even if you read 0 bytes because the end-of-file has been reached.
                 * Return true even if you read 0 bytes because data is not currently available.
                 *
                 * Return false if an actual failure condition occurs,
                 * you SHOULD also raise an error via aws_raise_error().
```

Reply:
updated